components/salsa-macro-rules/src/setup_tracked_fn.rs

```diff
@@ -284,9 +291,15 @@ macro_rules! setup_tracked_fn {
                 }
 
                 $zalsa::macro_if! { if0 $lru { } else {
+                    /// Sets the lru capacity
+                    ///
+                    /// **WARNING:** Just like an ordinary write, this method triggers
+                    /// cancellation. If you invoke it while a snapshot exists, it
+                    /// will block until that snapshot is dropped -- if that snapshot
+                    /// is owned by the current thread, this could trigger deadlock.
```

Comment:
I am a bit iffy about how our cancellation triggering functions all need this comment. I wonder if it would make sense to introduce a `CancellationToken` type (not to be confused with async cancellation tokens...) that one would need to supply to acknowledge the deadlock conditions as such an API forces the caller to think about this, where as the current status quo is having to have read the docs of the function.

Reply:
Right, that would infest all the input setters I believe hmm